BaseClasses.py

```diff
@@ -592,7 +592,7 @@ def not_flooding_a_key(self, world, location):
             flood_location = world.get_location(flooded_keys[location.name], location.player)
             item = flood_location.item
             item_is_important = False if not item else item.advancement or item.bigkey or item.smallkey
-            return flood_location in self.locations_checked or not item_is_important
+            return flood_location in self.locations_checked or not item_is_important or location.parent_region.name not in ['Swamp Trench 1 Alcove', 'Swamp Trench 2 Alcove']
```

Comment:
location should be flood_location and probably should add to the nearby function _do_not_flood_the_keys. I'll refactor it slightly and do some general testing before finished up this PR